web/src/OverviewItemView.tsx

```diff
@@ -0,0 +1,303 @@
+    }),
+  }).then((response) => {
+    if (!response.ok) {
+      console.log(response)
+    }
+  })
+}
+
+export type OverviewItemViewProps = {
+  item: OverviewItem
+  pathBuilder: PathBuilder
+}
+
+function buildStatusText(item: OverviewItem): string {
+  let buildDur = item.lastBuildDur ? formatBuildDuration(item.lastBuildDur) : ""
+  let buildStatus = item.buildStatus
+  if (buildStatus === ResourceStatus.Pending) {
+    return "Pending"
+  } else if (buildStatus === ResourceStatus.Building) {
+    return "Updating…"
+  } else if (buildStatus === ResourceStatus.Unhealthy) {
+    return "Update error"
+  } else if (buildStatus === ResourceStatus.Healthy) {
+    let msg = `Completed in ${buildDur}`
+    if (item.buildAlertCount > 0) {
+      msg += ", with issues"
+    }
+    return msg
+  }
+  return "Unknown"
+}
+
+function runtimeTooltipText(status: ResourceStatus): string {
+  switch (status) {
+    case ResourceStatus.Building:
+      return "Server: deploying"
+    case ResourceStatus.Pending:
+      return "Server: pending"
+    case ResourceStatus.Warning:
+      return "Server: issues"
+    case ResourceStatus.Healthy:
+      return "Server: ready"
+    case ResourceStatus.Unhealthy:
+      return "Server: unhealthy"
+    default:
+      return "No server"
+  }
+}
+
+function buildTooltipText(status: ResourceStatus): string {
+  switch (status) {
+    case ResourceStatus.Building:
+      return "Update: in progress"
+    case ResourceStatus.Pending:
+      return "Update: pending"
+    case ResourceStatus.Warning:
+      return "Update: warning"
+    case ResourceStatus.Healthy:
+      return "Update: success"
+    case ResourceStatus.Unhealthy:
+      return "Update: error"
+    default:
+      return "No update status"
+  }
+}
+
+export default function OverviewItemView(props: OverviewItemViewProps) {
+  let item = props.item
+  let link = `/r/${item.name}/overview`
+  let formatter = timeAgoFormatter
+  let hasSuccessfullyDeployed = !isZeroTime(item.lastDeployTime)
+  let hasBuilt = item.lastBuild !== null
+  let building = !isZeroTime(item.currentBuildStartTime)
+  let timeAgo = <TimeAgo date={item.lastDeployTime} formatter={formatter} />
+
+  let isBuildingClass = building ? "isBuilding" : ""
+  let onTrigger = triggerUpdate.bind(null, item.name)
+
+  return (
+    <OverviewItemRoot key={item.name} className={`${isBuildingClass}`}>
+      <OverviewItemBox
+        className={`${isBuildingClass}`}
+        to={props.pathBuilder.path(link)}
+        data-name={item.name}
+      >
+        <OverviewItemRuntimeBox>
+          <SidebarIcon
+            tooltipText={runtimeTooltipText(item.runtimeStatus)}
+            status={item.runtimeStatus}
+            alertCount={item.runtimeAlertCount}
+          />
+          <OverviewItemName name={item.name} />
+          <OverviewItemTimeAgo>
+            {hasSuccessfullyDeployed ? timeAgo : "—"}
+          </OverviewItemTimeAgo>
+          <SidebarTriggerButton
```

Comment:
The Trigger button shouldn't propagate clicks to its parents. Not blocking of this PR, but we should address it at some point.

Reply:
yep, there's a PR out for that! https://github.com/tilt-dev/tilt/pull/4003